components/net/http_loader.rs

```diff
@@ -311,30 +311,47 @@ impl HttpRequest for WrappedHttpRequest {
 pub struct LoadError {
     pub url: Url,
     pub error: LoadErrorType,
-    pub reason: String,
 }
 
 impl LoadError {
-    pub fn new(url: Url, error: LoadErrorType, reason: String) -> LoadError {
+    pub fn new(url: Url, error: LoadErrorType) -> LoadError {
         LoadError {
             url: url,
             error: error,
-            reason: reason,
         }
     }
 }
 
 #[derive(Eq, PartialEq, Debug)]
 pub enum LoadErrorType {
     Cancelled,
-    Connection,
-    ConnectionAborted,
-    Cors,
-    Decoding,
-    InvalidRedirect,
+    Connection { reason: String },
+    ConnectionAborted { reason: String },
+    // Preflight fetch inconsistent with main fetch
+    CorsPreflightFetchInconsistent,
+    Decoding { reason: String },
+    InvalidRedirect { reason: String },
     MaxRedirects(u32), // u32 indicates number of redirects that occurred
-    Ssl,
-    UnsupportedScheme,
+    RedirectLoop,
+    Ssl { reason: String },
+    UnsupportedScheme { scheme: String },
+}
+
+impl LoadErrorType {
+    fn error_reason(self) -> String {
+        match self {
+            LoadErrorType::Cancelled => "load cancelled".into(),
+            LoadErrorType::Connection { reason } => reason,
+            LoadErrorType::ConnectionAborted { reason } => reason,
+            LoadErrorType::CorsPreflightFetchInconsistent => "preflight fetch inconsistent with main fetch".into(),
+            LoadErrorType::Decoding { reason } => reason,
+            LoadErrorType::InvalidRedirect { reason } => reason,
+            LoadErrorType::MaxRedirects(_) => "too many redirects".into(),
+            LoadErrorType::RedirectLoop => "redirect loop".into(),
+            LoadErrorType::Ssl { reason } => reason,
+            LoadErrorType::UnsupportedScheme { .. } => "unsupported url scheme".into(),
+        }
+    }
```

Comment:
Yep, I don't like this `match` either. I'm pretty sure we can refactor it away in the future.